src/app/static/filtered-question-list/filtered-question-list.component.ts

```diff
@@ -123,7 +125,25 @@ export class FilteredQuestionListComponent implements OnInit {
   ngOnInit() {
     this.questionService.getQuestions().subscribe(unfilteredQuestions => {
       this.questions = unfilteredQuestions;
-
     });
+    this.selectedQuestionId = null;
+  }
+
+  ngOnChanges() {
+    if(this.selectedQuestionId) {
+      this.questionService.getQuestionById(this.selectedQuestionId).subscribe((data)=>{
+        this.selectedQuestion = data;
+      });
+      console.log(this.selectedQuestion);
+    } else {
+      console.log("onChange fired without right id.")
+    }
+  }
+
+  questionChange(value: number) {
+    console.log("ID CHANGED TO " + value + " from " + this.selectedQuestionId);
+    this.selectedQuestionId = value;
+    this.ngOnChanges();
```

Comment:
Instead of manually calling the lifecycle hook, split the functionality you want out into another method and call that both here and in the hook